docs/patterns/components/Pagination/index.js

```diff
@@ -1,19 +1,16 @@
 import React, { useState } from 'react';
 import Demo from '../../../Demo';
-import { Pagination } from '@deque/cauldron-react/';
+import { Pagination, usePagination } from '@deque/cauldron-react/';
 import { children, className } from '../../../props';
 
 const PaginationDemo = () => {
   const totalItems = 111;
   const itemsPerPage = 25;
-  const [currentPage, setCurrentPage] = useState(3);
-  const itemStart = currentPage * itemsPerPage - itemsPerPage + 1;
-  const itemEnd = Math.min(itemStart + itemsPerPage - 1, totalItems);
 
-  const onNextPageClick = () => setCurrentPage(currentPage + 1);
-  const onFirstPageClick = () => setCurrentPage(1);
-  const onPreviousPageClick = () => setCurrentPage(currentPage - 1);
-  const onLastPageClick = () => setCurrentPage(5);
+  const [pagination, pageStatus] = usePagination({
```

Comment:
Nitpick: I wonder if we should be more implicit in the naming of the return values here? I'd be concerned with someone confusing this to something like a `useState` which typically has `[state, setter]` in that order.

I think named values might be cleaner and more readable:

```ts
const { paginationState, paginationProps } = usePagination(...)
```

With named values, it would be more clear what's supposed to be spread onto the pagination component vs someone getting the order wrong for the prior example. What do you think?

Reply:
Yeah, that makes sense.